pkg/controller/registry/resolver/resolver.go

```diff
@@ -51,13 +52,15 @@ func NewOperatorsV1alpha1Resolver(lister operatorlister.OperatorLister, client v
 }
 
 func (r *OperatorsV1alpha1Resolver) ResolveSteps(namespace string, sourceQuerier SourceQuerier) ([]*v1alpha1.Step, []v1alpha1.BundleLookup, []*v1alpha1.Subscription, error) {
+	resolutionStartTime := time.Now()
```

Comment:
Instead of reporting a metric at every return site, and risking forgetting to update one in the future, the `error` return value could be named so that a single func to emit an error could be deferred here.

Alternatively, you could implement the `Resolver` interface with a type that wraps another `Resolver` and emits metrics, like this:
```
type InstrumentedResolver struct {
  wrapped Resolver
}

func (ir InstrumentedResolver) ResolveSteps(namespace string, sourceQuerier SourceQuerier) ([]*v1alpha1.Step, []v1alpha1.BundleLookup, []*v1alpha1.Subscription, error) {
  start := time.Now()
  steps, lookups, subs, err := ir.wrapped.ResolveSteps(namespace, sourceQuerier)
  elapsed := time.Now().Sub(start)
  if err != nil {
    metrics.RegisterDependencyResolutionFailure(elapsed.Seconds())
  } else {
    metrics.RegisterDependencyResolutionSuccess(elapsed.Seconds())
  }
  return steps, lookups, subs, err
}
```
The second option (especially if a metric reporter can be injected) would make it easy to test the instrumentation in isolation from resolution.